libs/galley-types/src/Galley/Types/Teams.hs

```diff
@@ -220,15 +219,15 @@ data FeatureFlags = FeatureFlags
   { _flagSSO :: !FeatureSSO,
```

Comment:
I think there is some redundancy between this type and `AllFeatureFlags` now, right?  Can we move this one to `wire-api` and use it instead of the other one?

I also want to make the type polymorphic in `WithLockStatus`.  Not sure we need the flexibility, but it would be nice to express more clearly that the value is the same throughout all the record fields.  Not sure it's a good idea, though.  If you do it for my reason, please explain in haddocks.

Reply:
That would be nice indeed, and it would work for SSO and SearchVilisibility, but AFAICT not easily for LegalHold as this config has more states than just enabled and disabled. Maybe we can convert it to a team features with config, though. I will try.